lib/webmushra/pages/SpatialPage.js

```diff
@@ -0,0 +1,382 @@
+};
+
+SpatialPage.prototype.render = function(_id) {
+
+	var div = document.createElement("div");
+
+	// content
+	var content;
+	if (this.pageConfig.content !== undefined) {
+		content = this.pageConfig.content;
+		var text = document.createElement("div");
+		text.innerHTML = content;
+		div.appendChild(text);
+	}
+	_id.append(div);
+
+	// stimuli
+	var tableStimuli = document.createElement("table");
+	tableStimuli.setAttribute("border", "0");
+	tableStimuli.setAttribute("align", "center");
+	tableStimuli.setAttribute("style", "margin-top: 0em;");
+	var trStimuli = document.createElement("tr");
+	for (var i = 0; i < this.stimuli.length; ++i) {
+		var td = $("<td></td>");
+		var divStimulus = $("<div id='spatial_stimuli_" + i + "' class='ui-body ui-body-a ui-corner-all'></div>");
+		td.append(divStimulus);
+		this.filePlayer.renderElement(divStimulus, i);
+
+		trStimuli.appendChild(td.get(0));
+	}
+	tableStimuli.appendChild(trStimuli);
+	div.appendChild(tableStimuli);
+
+	// responses
+	//var tableButtons = $("<table>");
+	//var trButton = ("tr");
+
+	for (var i = 0; i < this.pageConfig.responses.length; ++i) {
+		var respConfig = this.pageConfig.responses[i];
+		var stimulus = null;
+		var stimulusIndex = null;
+		for (var j = 0; j < this.stimuli.length; ++j) {
+			if (this.stimuli[j].getId() == respConfig.stimulus) {
+				stimulus = this.stimuli[j];
+				stimulusIndex = j;
+				break;
+			}
+		}
+
+		var color = "#" + ((1 << 24) + respConfig.color).toString(16).substr(1);
+		button = $("<button id='" + respConfig.name + "' data-inline='true' style='background-color:#" + ((1 << 24) + respConfig.color).toString(16).substr(1) + "; background-image:radial-gradient(#ffffff 0%, " + color + " 100%)'>" + respConfig.label + "</button>");
+		button.on("click", (function() {
+			this.page.setBoxShadow(this.page.activeResponse, this.name, "#555555");
+			this.page.activeResponse = this.name;
+		}).bind({
+			page : this,
+			label : respConfig.label,
+			name : respConfig.name,
+			button : button
+		}));
+		$("#spatial_stimuli_" + stimulusIndex).append(button);
+	}
+	// scenes
+	var tableScenes = document.createElement("table");
+	tableScenes.setAttribute("border", "0");
+	tableScenes.setAttribute("align", "center");
+	tableScenes.setAttribute("style", "margin-top: 0em;");
+	var trScenes = document.createElement("tr");
+	for (var i = 0; i < this.pageConfig.views.length; ++i) {
+		var td = document.createElement("td");
+		td.appendChild(this.renderDivs[i]);
+		trScenes.appendChild(td);
+	}
+
+	tableScenes.appendChild(trScenes);
+	div.appendChild(tableScenes);
+
+	this.renderScene();
+
+	_id.append(div);
+};
+
+SpatialPage.prototype.setBoxShadow = function(_lastActiveButton, _currentActiveButton, color) {
+	if (_lastActiveButton != null) {
+		document.getElementById(_lastActiveButton).style.boxShadow = "0 0 0 white , 0 0 0 white";
+
+	}
+	document.getElementById(_currentActiveButton).style.boxShadow = "0 0 30px " + color + " , 0 0 30px " + color;

```

Comment:
Probably yes. I guess, since the color can be freely configured, this design decision was chosen.